zerver/lib/upload.py

```diff
@@ -387,9 +387,9 @@ def __init__(self) -> None:
         self.uploads_bucket = get_bucket(settings.S3_AUTH_UPLOADS_BUCKET, self.session)
 
         self._boto_client = None
-        self.public_upload_url_pattern = self.construct_public_upload_url_pattern()
+        self.public_upload_url_base_parsed = self.construct_public_upload_url_base_parsed()
 
-    def construct_public_upload_url_pattern(self) -> str:
+    def construct_public_upload_url_base_parsed(self) -> urllib.parse.ParseResult:
```

Comment:
The other part of my comment was that we should deduplicate this function with `get_public_upload_root_url`, which does the same thing but more carefully.

Reply:
Yeah I missed that part somehow, sorry about that. Pushed again deduplicating that logic and taking the approach to parsing the url from `get_public_upload_root_url`